aptos-move/aptos-release-builder/src/components/mod.rs

```diff
@@ -77,9 +69,99 @@ impl ReleaseConfig {
                 .map_err(|err| anyhow!("Failed to write to file: {:?}", err))?;
         }
+
         Ok(())
     }
 
+    fn generate_framework_release(&self, result: &mut Vec<(String, String)>) {
+        let execution_hash = &Self::get_execution_hash(result);
+        if self.framework_release {
+            result.append(
+                &mut framework::generate_upgrade_proposals(
+                    self.testnet,
+                    if self.is_multi_step {
+                        execution_hash
+                    } else {
+                        ""
+                    },
+                )
+                .unwrap(),
+            );
+        }
+    }
+
+    fn generate_gas_schedule(&self, result: &mut Vec<(String, String)>) {
+        let execution_hash = &Self::get_execution_hash(result);
+        if let Some(gas_schedule) = &self.gas_schedule {
+            result.append(
+                &mut gas::generate_gas_upgrade_proposal(
+                    gas_schedule,
+                    self.testnet,
+                    if self.is_multi_step {
+                        execution_hash
+                    } else {
+                        ""
+                    },
+                )
+                .unwrap(),
+            );
+        }
+    }
+
+    fn generate_version_file(&self, result: &mut Vec<(String, String)>) {
+        let execution_hash = &Self::get_execution_hash(result);
+        if let Some(version) = &self.version {
+            result.append(
+                &mut version::generate_version_upgrade_proposal(
+                    version,
+                    self.testnet,
+                    if self.is_multi_step {
+                        execution_hash
+                    } else {
+                        ""
+                    },
+                )
+                .unwrap(),
+            );
+        }
+    }
+
+    fn generate_feature_flag_file(&self, result: &mut Vec<(String, String)>) {
+        let execution_hash = &Self::get_execution_hash(result);
+        if let Some(feature_flags) = &self.feature_flags {
+            result.append(
+                &mut feature_flags::generate_feature_upgrade_proposal(
+                    feature_flags,
+                    self.testnet,
+                    if self.is_multi_step {
+                        execution_hash
+                    } else {
+                        ""
+                    },
+                )
+                .unwrap(),
+            );
+        }
+    }
+
+    fn generate_consensus_file(&self, result: &mut Vec<(String, String)>) {
+        let execution_hash = &Self::get_execution_hash(result);
+        if let Some(consensus_config) = &self.consensus_config {
+            result.append(
+                &mut consensus_config::generate_consensus_upgrade_proposal(
+                    consensus_config,
+                    self.testnet,
+                    if self.is_multi_step {
```

Comment:
Why not move this up to line 148? Same for earlier functions

Reply:
I tried doing
```        
let execution_hash = if  { &Self::get_execution_hash(result) } else {""};
```
but it keeps giving me this temporary value error

```
77 |         let execution_hash = if self.is_multi_step {
   |             -------------- borrow later stored here
78 |             &Self::get_execution_hash(result).to_string()
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ creates a temporary which is freed while still in use
79 |         } else {
   |         - temporary value is freed at the end of this statement
   |
   = note: consider using a `let` binding to create a longer lived value
```
so I just do the `if-else` statement in the function call - maybe a bit hacky, lmk if there's a better way to do it : )